src/hotspot/share/logging/logAsyncWriter.cpp

```diff
@@ -152,9 +155,16 @@ void AsyncLogWriter::write() {
 
 void AsyncLogWriter::run() {
   while (true) {
-    // The value of a semphore cannot be negative. Therefore, the current thread falls asleep
-    // when its value is zero. It will be waken up when new messages are enqueued.
-    _sem.wait();
+    {
+      AsyncLogLocker locker;
+
+      while (!_data_available) {
+        _lock.signal();
+        _cv.wait(0/* no timeout */);
```

Comment:
You can't call wait() on an unlocked monitor  - that's not how monitors are used.